iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -187,6 +187,146 @@ struct SliceOpConversion : public OpConversionPattern<mhlo::SliceOp> {
 };
 }  // namespace
 
+//===----------------------------------------------------------------------===//
+// mhlo.conv conversion patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+bool isDepthwiseConv(mhlo::ConvOp op) {
+  auto shape = op.rhs().getType().cast<ShapedType>().getShape();
+  auto numGroups =
+      shape[op.dimension_numbers().kernel_input_feature_dimension().getInt()];
+  return op.feature_group_count() > 1u && op.feature_group_count() == numGroups;
+}
+
+/// Converts mhlo.conv operation to linalg named op.
+struct ConvOpConversion : public OpConversionPattern<mhlo::ConvOp> {
+  using OpConversionPattern<mhlo::ConvOp>::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      mhlo::ConvOp op, ArrayRef<Value> args,
+      ConversionPatternRewriter &rewriter) const override {
+    if (isDepthwiseConv(op)) return failure();
+
+    mhlo::ConvOp::Adaptor adaptor(args);
+    Location loc = op.getLoc();
+    Value input = adaptor.lhs();
+    Value filter = adaptor.rhs();
+    if (const auto dimensionNumbers = op.dimension_numbers()) {
```

Comment:
Ooh, I think this is gonna collide with a change I've: https://github.com/google/iree/pull/4799. I'm extracting the `dimensionNumbers` check there into a function. Would you mind to just copy the changes there so that we can reduce the workload of rebasing? 

Also it would be nice if you can convert the tensor version of depthwise conv to memref version too. :) (But not required for this change.)

Reply:
I think your patch is ready to land. I will wait that and rebase and fix it. Thanks for the heads-up.